.buildkite/pipeline.yml

```diff
@@ -1,93 +1,154 @@
+container:
+  kubernetes: &kubernetes
+    gitEnvFrom:
+      - secretRef:
+          name: oss-github-ssh-credentials
+    sidecars:
+    - image: us-west1-docker.pkg.dev/ci-compute/buildkite-images/buildkite-dind:v1
+      volumeMounts:
+        - mountPath: /var/run/
+          name: docker-sock
+      securityContext:
+        privileged: true
+        allowPrivilegeEscalation: true
+    mirrorVolumeMounts: true # CRITICAL: this must be at the same indentation level as sidecars
+    podSpec: &podSpec
+      containers:
+        - &commandContainer
+          image: us-west1-docker.pkg.dev/ci-compute/buildkite-images/buildkite-command-container:v2
+          command:
+          - |-
+            echo "Command step was not overridden."
+            exit 1
+          volumeMounts:
+            - mountPath: /var/run/
+              name: docker-sock
+          resources:
+            requests:
+              cpu: 7500m
+              memory: 28G
+      volumes:
+      - name: docker-sock
+        emptyDir: {}
+
+agents:
+  queue: "buildkite-gcp"
 steps:
   - label: "fossa analyze"
-    agents:
-      queue: "init"
-      docker: "*"
     command: ".buildkite/scripts/fossa.sh"
 
   - label: ":golang: unit-test"
-    agents:
-      queue: "workers"
-      docker: "*"
-    commands:
-      - "make unit_test"
-      - ".buildkite/scripts/gen_coverage_metadata.sh .build/metadata.txt"
     artifact_paths:
       - ".build/*/coverage/*.out"
       - ".build/cover.out"
       - ".build/metadata.txt"
     plugins:
+      - kubernetes:
+          <<: *kubernetes
```

Comment:
@mstifflin your yaml fu is way beyond mine, I assume this means destructuring the k8s block defined ?

Reply:
I definitely learned something new working on this. This syntax basically allows me to overwrite **just** the command step in that nested kubernetes block.